nengo_loihi/splitter.py

```diff
@@ -4,184 +4,280 @@
+            ):
+                self.host_objs.add(ens)
+            else:
+                self.chip_objs.add(ens)
+
+        for conn in network.all_connections:
+            pre, post = base_obj(conn.pre), base_obj(conn.post)
+
+            # Enforce rule 3
+            if (
+                conn.learning_rule_type is not None
+                and isinstance(post, Ensemble)
+                and post in self.chip_objs
+            ):
+                if network.config[post].on_chip:
+                    raise BuildError(
+                        "Post ensemble (%r) of learned connection (%r) must not be "
+                        "configured as on_chip." % (post, conn)
+                    )
+                self.host_objs.add(post)
+                self.chip_objs.remove(post)
+
+            # Enforce rule 4
+            elif (
+                isinstance(post, LearningRule)
+                and isinstance(pre, Ensemble)
+                and pre in self.chip_objs
+            ):
+                if network.config[pre].on_chip:
+                    raise BuildError(
+                        "Pre ensemble (%r) of error connection (%r) must not be "
+                        "configured as on_chip." % (pre, conn)
+                    )
+                self.host_objs.add(pre)
+                self.chip_objs.remove(pre)
+
+    def _place_probes(self, network):
+        """Place probes. Probes go where their probed object is."""
+
+        for probe in network.all_probes:
+            obj = base_obj(probe.target)
+            if obj in self.host_objs:
+                self.host_objs.add(probe)
+            elif obj in self.chip_objs:
+                self.chip_objs.add(probe)
+            else:
+                raise BuildError("Object (%r) is not a part of the network" % (obj,))
 
     def on_chip(self, obj):
-        if isinstance(obj, Probe):
-            obj = base_obj(obj.target)
-        if not isinstance(obj, (Ensemble, Node)):
+        if not isinstance(obj, (Ensemble, Node, Probe)):
             raise BuildError(
                 "Locations are only established for ensembles ",
                 "nodes, and probes -- not for %r" % (obj,),
             )
-        if obj not in self._seen_objects:
-            raise BuildError("Object (%r) is not a part of the network" % (obj,))
-        return obj in self._chip_objects
+        if obj in self.chip_objs:
+            return True
+        elif obj in self.host_objs:
+            return False
+        raise BuildError("Object (%r) is not a part of the network" % (obj,))
+
+
+class Split:
+    """Creates a set of directives to guide the builder.
+
+    Parameters
+    ----------
+    network : Network
+        The original Network supplied to the builder.
+    precompute : bool, optional (Default: None)
+        Whether model inputs should be precomputed to speed up simulation.
+        The splitter will determine precomputable objects regardless,
+        but if
```

Comment:
Docstring stops abruptly. Attributes also empty.

Reply:
Fixed in 3ffed333